### PR TITLE
A few fixes of the deprecations of the keyword versions of cov and cor.

### PR DESCRIPTION
**Pull request by andreasnoack**
Also make sure that all methods are tested.

### CODE REVIEW

**Comment by jiahao** [comment]
Confirming that the tests of `MultivariateStats.jl` now pass again on this branch. But `StatsBase.jl` is still failing because `covm` is not exported. The deprecation rule should reference `Base.covm` instead.

**Comment by jiahao** [comment]
@andreasnoack is the issue with StatsBase.jl fixed now?

**Comment by andreasnoack** [comment]
I think so, but we'll have to wait for new nightlies before the StatsBase tests will pass.

File: base/deprecated.jl [hunk 1]
```diff
@@ -843,12 +843,12 @@ for f in (:remotecall, :remotecall_fetch, :remotecall_wait)
...
-@deprecate cov(x::AbstractVector; corrected=true, mean=nothing) covm(x, mean, corrected)
-@deprecate cov(X::AbstractMatrix; vardim=1, corrected=true, mean=nothing) covm(X, mean, vardim, corrected)
-@deprecate cov(x::AbstractVector, y::AbstractVector; corrected=true, mean=nothing) covm(x, mean[1], y, mean[2], corrected)
-@deprecate cov(X::AbstractVecOrMat, Y::AbstractVecOrMat; vardim=1, corrected=true, mean=nothing) covm(X, mean[1], Y, mean[2], vardim, corrected)
-
-@deprecate cor(x::AbstractVector; mean=nothing) corm(x, mean)
-@deprecate cor(X::AbstractMatrix; vardim=1, mean=nothing) corm(X, mean, vardim)
-@deprecate cor(x::AbstractVector, y::AbstractVector; mean=nothing) corm(x, mean[1], y, mean[2])
-@deprecate cor(X::AbstractVecOrMat, Y::AbstractVecOrMat; vardim=1, mean=nothing) corm(X, mean[1], Y, mean[2], vardim)
+@deprecate cov(x::AbstractVector; corrected=true, mean=Base.mean(x)) covm(x, mean, corrected)
```
Comment (stevengj) [inline base/deprecated.jl]: This deprecation is wrong because `covm` is not yet exported.

**Pull request merged**